tests/tensorboard_test.py

```diff
@@ -0,0 +1,204 @@
+
+from tensorboard.backend.event_processing import directory_watcher
+from tensorboard.backend.event_processing import event_file_loader
+from tensorboard.util import tensor_util
+import tensorflow.compat.v2 as tf
+
+from flax.metrics.tensorboard import SummaryWriter
+
+def _process_event(event):
+  for value in event.summary.value:
+    yield {'wall_time': event.wall_time, 'step': event.step, 'value': value}
+
+
+class TensorboardTest(absltest.TestCase):
+
+  def parse_and_return_summary_value(self, path):
+    """Parse the event file in the given path and return the
+    only summary value."""
+    event_value_list = []
+    event_file_generator = directory_watcher.DirectoryWatcher(
+        path, event_file_loader.EventFileLoader).Load()
+    event_values = itertools.chain.from_iterable(
+        map(_process_event, event_file_generator))
+    for value_dict in event_values:
+      event_value_list.append(value_dict)
+
+    self.assertLen(event_value_list, 1)
+    self.assertEqual(event_value_list[0]['step'], 1)
+    self.assertGreater(event_value_list[0]['wall_time'], 0.0)
+    return event_value_list[0]['value']
+
+  def assert_encoded_audio_files_equal(self, actual_audio, expected_audio):
+    # expected_audio is trimmed to -1.0 to 1.0
+    self.assertFalse(onp.allclose(actual_audio.numpy(), expected_audio.numpy()))
+
+    # trim the audio to assert the values.
+    trimmed_audio = onp.clip(onp.array(actual_audio), -1, 1)
+    self.assertTrue(
+        onp.allclose(trimmed_audio, expected_audio.numpy(), atol=1e-04))
+
+  def test_summarywriter_flush_after_close(self):
+    log_dir = tempfile.mkdtemp()
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    summary_writer.close()
+    with self.assertRaises(AttributeError):
+      summary_writer.flush()
+
+  def test_summarywriter_scalar(self):
+    log_dir = tempfile.mkdtemp()
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    # Write the scalar and check if the event exists and check data.
+    float_value = 99.1232
+    summary_writer.scalar(tag='scalar_test', value=float_value, step=1)
+
+    summary_value = self.parse_and_return_summary_value(path=log_dir)
+    self.assertEqual(summary_value.tag, 'scalar_test')
+    self.assertTrue(onp.allclose(
+        tensor_util.make_ndarray(summary_value.tensor).item(),
+        float_value))
+
+  def test_summarywriter_text(self):
+    log_dir = tempfile.mkdtemp()
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    text = 'hello world.'
+    summary_writer.text(tag='text_test', textdata=text, step=1)
+
+    summary_value = self.parse_and_return_summary_value(path=log_dir)
+    self.assertEqual(summary_value.tag, 'text_test')
+    self.assertEqual(
+        tensor_util.make_ndarray(summary_value.tensor).item().decode('utf-8'),
+        text)
+
+  def test_summarywriter_image(self):
+    log_dir = tempfile.mkdtemp()
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    img = tf.random.uniform(shape=[30, 30, 3])
```

Comment:
Why are these TF tensors and not numpy arrays?

Reply:
Same comment also below. Shouldn't the summary writer get numpy arrays or JAX DeviceArrays?

Reply:
Thanks for catching this. Yes, these should be ndarrays as mentioned in the docstrings of the methods. 

Changed the input for image and audio both.